_overviews/core/collections-migration-213.md

```diff
@@ -0,0 +1,330 @@
+
+`scala.collection.immutable.ArraySeq` is an immutable sequence backed by an array. It is used to pass varargs parameters.
+
+The [`scala-collection-contrib`](https://github.com/scala/scala-collection-contrib) module provides decorators enriching the collections with new operations. You can
+think of this artifact as an incubator: if we get evidence that these operations should be part of the core,
+we might eventually move them.
+
+The following collections are provided:
+
+- `MultiSet` (both mutable and immutable)
+- `SortedMultiSet` (both mutable and immutable)
+- `MultiDict` (both mutable and immutable)
+- `SortedMultiDict` (both mutable and immutable)
+
+## Are there new operations on collections?
+
+The following new partitioning operations are available:
+
+~~~ scala
+def groupMap[K, B](key: A => K)(f: A => B): Map[K, CC[B]] // (Where `CC` can be `List`, for instance)
+def groupMapReduce[K, B](key: A => K)(f: A => B)(g: (B, B) => B): Map[K, B]
+~~~
+
+`groupMap` is equivalent to `groupBy(key).mapValues(_.map(f))`.
+
+`groupMapReduce` is equivalent to `groupBy(key).mapValues(_.map(f).reduce(g))`.
+
+Mutable collections now have transformation operations that modify the collection in place:
+
+~~~ scala
+def mapInPlace(f: A => A): this.type
+def flatMapInPlace(f: A => IterableOnce[A]): this.type
+def filterInPlace(p: A => Boolean): this.type
+def patchInPlace(from: Int, patch: scala.collection.Seq[A], replaced: Int): this.type
+~~~
+
+Another new operation is `distinctBy`:
+
+~~~ scala
+def distinctBy[B](f: A => B): C // (Where `C` can be `List[Int]`, for instance)
+~~~
+
+Last, additional operations are provided by the `scala-collection-contrib` module. You can
+think of this artifact as an incubator: if we get evidence that these operations should be part of the core,
+we might eventually move them.
+
+The new operations are provided via an implicit enrichment. You need to add the following import to make them
+available:
+
+~~~ scala
+import strawman.collection.decorators._
+~~~
+
+The following operations are provided:
+
+- `Seq`
+    - `intersperse`
+- `Map`
+    - `zipByKey` / `join` / `zipByKeyWith`
+    - `mergeByKey` / `fullOuterJoin` / `mergeByKeyWith` / `leftOuterJoin` / `rightOuterJoin`
+
+## Are there new implementations of existing collection types (changes in performance characteristics)?
+
+The default `Set` and `Map` are backed by a `ChampHashSet` and a `ChampHashMap`, respectively. The performance characteristics are the same but the
+operation implementations are faster. These data structures also have a lower memory footprint.
+
+`mutable.Queue` and `mutable.Stack` now use `mutable.ArrayDeque`. This data structure supports constant time index access, and amortized constant time
+insert and remove operations.
+
+## How do I cross-build my project against Scala 2.12 and Scala 2.13?
+
+Most usages of collections are compatible and can cross-compile 2.12 and 2.13 (at the cost of some warnings, sometimes).
```

Comment:
We could maybe remove this line?

Reply:
I left it, seems useful for people just reading the doc, not yet doing any coding.